### PR TITLE
Feature/sscs 2817 update events

### PR DESCRIPTION
**Pull request by iguannago**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-2817

### Change description ###
Major refactoring to make classes smaller and simpler to unit test in general. 
Add tests and logic to update cases only when there is a difference in the counting of new events come in and events already in CDD,

**Does this PR introduce a breaking change?** (check one with "x")

```
[ x] Yes
[ ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSenderTest.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+            .build();
+
+        ccdCasesSender.sendUpdateCcdCases(Collections.singletonList(caseData));
+
+        verify(updateCoreCaseDataService, times(0))
+            .updateCase(eq(caseData), anyLong(), any());
+    }
+
+    @Test
+    public void givenNewEventIseNull_shouldNotUpdateCcd() throws Exception {
```
Comment (jackmaloney) [inline src/test/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSenderTest.java]: extra e in method name

File: src/test/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSenderTest.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+    private CaseDetails getCaseDetails(String caseDetails) throws Exception {
+        String caseDetailsJson = FileUtils.readFileToString(new File(caseDetails),
+            StandardCharsets.UTF_8.name());
+        ObjectMapper mapper = Jackson2ObjectMapperBuilder.json().build();
+        return mapper.readerFor(CaseDetails.class).readValue(caseDetailsJson);
+    }
+
+    @Test
+    @Parameters({"APPEAL_RECEIVED", "RESPONSE_RECEIVED", "HEARING_BOOKED", "HEARING_POSTPONED", "APPEAL_LAPSED",
```
Comment (jackmaloney) [inline src/test/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSenderTest.java]: Is it really necessary to test that there is no event change for every single event type? Surely once is good enough?

File: src/test/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSenderTest.java (hunk 1 above)
Comment (iguannago) [reply src/test/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSenderTest.java]: thanks

File: src/test/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSenderTest.java (hunk 2 above)
Comment (iguannago) [reply src/test/java/uk/gov/hmcts/reform/sscs/services/ccd/CcdCasesSenderTest.java]: yeah, right. Thanks

**Pull request merged**